pkg/oci/oci.go

```diff
@@ -478,16 +490,85 @@ func newAuthClient(repository string, authOptions *AuthOptions) (*oras_auth.Clie
 	}, nil
 }
 
+func prepareCheckOpts(ctx context.Context, publicKey string) (*cosign.CheckOpts, error) {
+	pubKey, err := signature.LoadPublicKeyRaw([]byte(publicKey), crypto.SHA256)
+	if err != nil {
+		return nil, fmt.Errorf("loading public key: %w", err)
+	}
+
+	rekorPubKeys, err := cosign.GetRekorPubs(ctx)
+	if err != nil {
+		return nil, fmt.Errorf("getting Rekor public keys: %w", err)
+	}
+
+	cTLogPubKeys, err := cosign.GetCTLogPubs(ctx)
+	if err != nil {
+		return nil, fmt.Errorf("getting ctlog public keys: %w", err)
+	}
+
+	return &cosign.CheckOpts{
+		SigVerifier:  pubKey,
+		RekorPubKeys: rekorPubKeys,
+		CTLogPubKeys: cTLogPubKeys,
+	}, nil
+}
+
+func prepareImageReference(ctx context.Context, image string) (name.Reference, error) {
+	ref, err := name.ParseReference(image)
+	if err != nil {
+		return nil, fmt.Errorf("parsing reference: %w", err)
+	}
+
+	ref, err = sign.GetAttachedImageRef(ref, "", ociremote.WithRemoteOptions(googleremote.WithContext(ctx)))
+	if err != nil {
+		return nil, fmt.Errorf("resolving attachment for image %s: %w", image, err)
+	}
+
+	return ref, nil
+}
+
+func verifyImage(image string, publicKey string) error {
+	ctx := context.TODO()
+
+	co, err := prepareCheckOpts(ctx, publicKey)
+	if err != nil {
+		return fmt.Errorf("preparing check options: %w", err)
+	}
+
+	ref, err := prepareImageReference(ctx, image)
+	if err != nil {
+		return fmt.Errorf("preparing image reference: %w", err)
+	}
+
+	_, _, err = cosign.VerifyImageSignatures(ctx, ref, co)
```

Comment:
If I'm using an insecure registry, don't we need to specify it here somehow? Same for any other auth options like the authentication file? I don't know, maybe I'm very probably missing something 😅